### DISCUSSION TITLE
Building   jlama-native project on Windows 11

### DISCUSSION THREAD

**Discussion by przemSzer** [tjake/Jlama#31] [General]
Hi, 
**at first thank you very much, for this interesting JLama project**  👍 . 
I am a Java developer, and it is realy valuable for me to have some LLM written in Java, where I can easily review code and use tools which I am familiar with.

I wanted to get advise how do you build  jlama-native under Windows 11.

I use IntelliJ and maven. I had a problem that Make is not found:

`Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:1.8:run (build-native-lib) on project jlama-native: An Ant BuildException has occured: Execute failed: java.io.IOException: Cannot run program "make" (in directory "C:\development\szkolenia\Jlama\jlama-native"): CreateProcess error=2, Can not find given file
[ERROR] around Ant part ...<exec resolveexecutable="true" failonerror="true" executable="make">... @ 4:71 in C:\development\szkolenia\Jlama\jlama-native\target\antrun\build-main.xml
[ERROR] -> [Help 1]`.
 
I haven't installed anything yet to fix it. So far I skipped this project (and use tensor computations based on Java's panama).
I also know there are build jars in maven repo, but I want to build it myself.

Which tool you suggest to install in order to have make available (and build JLama-native)? WSL or bin shell, or maybe Visual Studio has some extension for make files?

**Comment by tjake** [comment]
Hi, 

Thanks for the feedback.  You can use the pre-built libraries in maven (See Readme)

Or I think you can use https://chocolatey.org/ to install gcc
